### PR TITLE
Vice: rewrite translator

### PR DESCRIPTION
**Pull request by qubidt**
Update the vice.com translator to work with the current incarnation of the site.

The multiple-page tests don't pass in scaffold, I assume because the results are pulled dynamically rather than as part of the initial GET. Not sure what to do there.

### CODE REVIEW

File: Vice.js [hunk 1]
```diff
@@ -148,32 +64,77 @@ function getSearchResults(doc, checkOnly) {
...
+		if (linkedData) {
+			linkedData = JSON.parse(linkedData);
+			if (linkedData && linkedData['@graph']) {
+				linkedData = linkedData['@graph'].find(x => x['@type'] == "NewsArticle");
+				if (linkedData) item.date = linkedData.datePublished;
+			}
+		}
+
+		item.creators.length = 0;
+		for (const author of doc.querySelectorAll('main.main-content .contributors .contributor .contributor__meta a')) {
```
Comment (AbeJellinek) [inline Vice.js:88]: Can we make this a bit shorter to be more resilient to site changes? Like:

```suggestion
		for (const author of doc.querySelectorAll('main .contributor__meta a')) {
```

It's less specific, but it still seems to match fine and it'll survive a reorganization of the site's outer layout elements.

File: Vice.js [hunk 2]
```diff
@@ -148,32 +64,77 @@ function getSearchResults(doc, checkOnly) {
...
+			item.creators.push(ZU.cleanAuthor(author.textContent, 'author'));
+		}
+		for (const author of attr(doc, 'meta[property="article:author"]', 'content').split(",")) {
+			item.creators.push(ZU.cleanAuthor(author, 'author'));
```
Comment (AbeJellinek) [inline Vice.js:92]: What happens if an author is in the `.contributors` and `article:author`?

**Review by AbeJellinek — CHANGES_REQUESTED** [review]
Thanks! A couple comments.

**Comment by AbeJellinek** [comment]
Oh, and you can add `"defer": true` to a test's JSON object to have Scaffold wait for the page to populate dynamically before running the test.

**Comment by qubidt** [comment]
> Oh, and you can add `"defer": true` to a test's JSON object to have Scaffold wait for the page to populate dynamically before running the test.

Brilliant, thank you. I'll test that out

File: Vice.js (hunk 2 above)
Comment (qubidt) [reply Vice.js:92]: Good catch. I hadn't encountered this but it doesn't hurt to be safe. I'll throw them in a Set

File: Vice.js (hunk 1 above)
Comment (qubidt) [reply Vice.js:88]: Good call, it seems to work well, I'll opt for that instead

Comment (AbeJellinek) [reply Vice.js:88]: Cool, sounds good.

**Comment by AbeJellinek** [comment]
No problem. It doesn't show in the Testing pane, but it'll work and be preserved between populates/saves fine.

**Comment by AbeJellinek** [comment]
Hey, I'd love to get this merged! Have you gotten a chance to finish it up?

**Comment by owcz** [comment]
Superseded by #2761?

**Pull request closed without merging**